docs/tutorials/services/constrain-motion.md

```diff
@@ -0,0 +1,433 @@
+In this tutorial, you'll expand on the code that describes your robot's working environment in two ways:
+
+- Define an additional obstacle: a tissue box sitting on the table.
+- Add a `z_offset` parameter to the `table_origin` and to the new `box_origin`.
+  This makes it easier to calibrate your motion plans based on how high or low your arm is mounted compared to the table.
+
+The following example shows the two obstacles defined with the expanded configuration.
+Adjust the dimensions and positions of the obstacles to describe your own scenario:
+
+```python {class="line-numbers linkable-line-numbers"}
+# Use this offset to set your z to calibrate based on where your table actually is
+z_offset = 10
+
+# Create a table obstacle
+table_origin = Pose(x=0.0, y=0.0, z=-19.0+z_offset)
+table_dims = Vector3(x=2000.0, y=2000.0, z=38.0)
+table_object = Geometry(center=table_origin, box=RectangularPrism(dims_mm=table_dims))
+
+# Create a tissue box obstacle
+box_origin = Pose(x=400, y=0, z=50+z_offset)
+box_dims = Vector3(x=120.0, y=80.0, z=100.0)
+box_object = Geometry(center=box_origin, box=RectangularPrism(dims_mm=box_dims))
+
+obstacles_in_frame = GeometriesInFrame(reference_frame="world",
+    geometries=[table_object, box_object])
+```
+
+## Determine the gripper's axes
+
+You need to identify how the gripper's frame corresponds to its hardware so that you can write code that creates your desired behavior.
+That is, you need to determine which axis points from the "wrist" to the end of the gripper, which axis lines up with the direction the jaws actuate, and, when the gripper is holding a cup, which axis passes vertically through the cup.
+
+All example code below assumes +Z points from the base of the gripper to the point where its jaws close, +Y points towards the ground when the gripper is holding a cup from the side, and the X axis is the axis along which the jaws close, following the [right-hand rule](https://en.wikipedia.org/wiki/Right-hand_rule).
+The following diagram shows this, as well as the global coordinate system.
+
+![A gripper mounted on an arm. The Z axis of the gripper points from the base of the gripper to the end of its jaws. The X axis points up through the gripper. The Y axis points in the direction along which the jaws open and close (following the right-hand rule). The diagram also shows the global coordinate system with Z pointing up, X down the length of the horizontal gripper, and Y pointing horizontally in the opposite direction of the gripper's Y.](../../img/constrain-motion/gripper-diagram.png)
+
+If you are using a `fake` gripper, there is no real hardware to calibrate and you can [continue this tutorial](#use-a-transform-to-represent-a-drinking-cup), imagining that your fake gripper corresponds to the diagram above.
```

Comment:
```suggestion
If you are using a `fake` gripper, there is no real hardware to calibrate and you can [continue to the next section](#use-a-transform-to-represent-a-drinking-cup), imagining that your fake gripper corresponds to the diagram above.
```

You're already in a tutorial so this makes it a bit more specific and reads better.

Reply:
True this is better